lib/perl/Genome/InstrumentData/AlignmentResult_regenerate.t

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env genome-perl
+
+use strict;
+use warnings FATAL => 'all';
+
+BEGIN {
+    $ENV{UR_DBI_NO_COMMIT} = 1;
+    $ENV{UR_USE_DUMMY_AUTOGENERATED_IDS} = 1;
+};
+
+use above "Genome";
+use Test::More;
+use Genome::Utility::Test;
+use Genome::Test::Factory::Test qw(test_setup_object);
+use Genome::Test::Factory::InstrumentData::AlignmentResult;
+use Genome::Test::Factory::InstrumentData::MergedAlignmentResult;
+use Genome::Test::Factory::DiskAllocation;
+use Sub::Override;
+use Test::MockObject::Extends;
+use Sub::Install;
+use Genome::Utility::Test qw(compare_ok);
+
+my $pkg = 'Genome::InstrumentData::AlignmentResult';
+use_ok($pkg) or die;
+
+my $merge_id       = 'whole';
+my $small_merge_id = 'small';
+my $test_name      = 'AlignmentResult_regenerate.t unit test objects';
+my $test_data_dir  = Genome::Utility::Test->data_dir_ok($pkg, '1');
+
+my $reference_build_id = 1234;
+my $ar1_instrument_data_id = 2894005119;
+my $ar2_instrument_data_id = 2894005341;
+my $per_lane_file_basename = 'all_sequences';
+my $per_lane_bam = $per_lane_file_basename . '.bam';
+my $per_lane_header = $per_lane_bam . '.header';
+my $per_lane_flagstat = $per_lane_bam . '.flagstat';
+
+my $ar_class         = 'Genome::Test::Factory::InstrumentData::AlignmentResult';
+my $merge_class      = 'Genome::Test::Factory::InstrumentData::MergedAlignmentResult';
+my $allocation_class = 'Genome::Test::Factory::DiskAllocation';
+
+# We need to override this because when an allocation is moved the contents do not seem to be copied. A bug or something we messed up? FIXME 
```

Comment:
I think this is now unnecessary. Tests seem to agree! Thanks so much for the catch!